elephant/spike_train_surrogates.py

```diff
@@ -46,25 +46,61 @@
 from __future__ import division, print_function, unicode_literals
 
 import random
+from functools import partial
 
 import neo
 import numpy as np
 import quantities as pq
 from scipy.ndimage import gaussian_filter
 
-try:
-    import elephant.statistics as es
-
-    isi = es.isi
-except ImportError:
-    from .statistics import isi  # Convenience when in elephant working dir.
+from elephant.statistics import isi
 
 # List of all available surrogate methods
 SURR_METHODS = ['dither_spike_train', 'dither_spikes', 'jitter_spikes',
-                'randomise_spikes', 'shuffle_isis', 'joint_isi_dithering']
+                'randomise_spikes', 'shuffle_isis', 'joint_isi_dithering',
+                'dither_spikes_with_refractory_period']
+
+
+def _dither_spikes_with_refractory_period(spiketrain, dither, n,
+                                          refractory_period):
+    units = spiketrain.units
+    t_start = spiketrain.t_start.rescale(units).magnitude
+    t_stop = spiketrain.t_stop.rescale(units).magnitude
+
+    dither = dither.rescale(units).magnitude
+    refractory_period = refractory_period.rescale(units).magnitude
+    refractory_period = np.min(np.diff(spiketrain.magnitude),
+                               initial=refractory_period)
+
+    dithered_spiketrains = []
+    for _ in range(n):
+        dithered_st = np.copy(spiketrain.magnitude)
+        random_ordered_ids = np.arange(len(spiketrain))
+        np.random.shuffle(random_ordered_ids)
+
+        for random_id in random_ordered_ids:
+            spike = dithered_st[random_id]
+            prev_spike = dithered_st[random_id - 1] \
+                if random_id > 0 \
+                else t_start - refractory_period
+            next_spike = dithered_st[random_id + 1] \
+                if random_id < len(spiketrain) - 1 \
+                else t_stop + refractory_period
+            prev_dither = min(dither, spike - prev_spike - refractory_period)
```

Comment:
I think that the word dither here is misleading...in principle it's just the space from your spike to the previous spike minus the refractory period, right? You are not actually dithering the previous spike if I understood correctly...

Reply:
I added a comment in the code. 
I stands for the dither range in the direction of the previous spike. This is either the dither parameter itself or the distance between the spikes minus the refractory period.